### PR TITLE
Extract elapsedTime for queries [SC-9645]

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

Extract the elapsed time of query for `BigQueryQueryExtractor`, `SnowflakeQueryExtractor` and `RedshiftQueryExtractor`.

### 🤓 What?

- Fill elapsedTime for 3 platform query extractor.
- Fix BigQuery usage & query crawler issue. `logEvent. from_entry` return `None` if destination table is not valid in that query.

### 🧪 Tested?

- Run 3 query extractor and BigQuery usage/lineage extractor.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -21,7 +21,7 @@ google-cloud-bigquery = { version = "^2.21.0", optional = true }
 google-cloud-logging = { version = "^2.7.0", optional = true }
```
Comment (mars-lan) [inline pyproject.toml:21]: Also bump up version?

Comment (elic-eon) [reply pyproject.toml:21]: version 3?

File: metaphor/bigquery/query/extractor.py [hunk 2]
```diff
@@ -95,9 +95,14 @@ def _parse_job_change_entry(self, entry: LogEntry) -> None:
 
             self._table_queries.store_recent_query(
                 table_name,
-                job_change.timestamp,
-                job_change.query,
-                job_change.user_email,
+                QueryInfo(
+                    elapsed_time=(
```
Comment (mars-lan) [inline metaphor/bigquery/query/extractor.py:99]: Make sure this returns float otherwise it'll throw conversion errors

File: pyproject.toml (hunk 1 above)
Comment (mars-lan) [reply pyproject.toml:21]: I mean the version for the whole project. GitHub doesn't allow me to add comments to non-changed lines.

Comment (elic-eon) [reply pyproject.toml:21]: Done. I thought I have bumped up the version 😵‍💫

File: metaphor/bigquery/query/extractor.py (hunk 2 above)
Comment (elic-eon) [reply metaphor/bigquery/query/extractor.py]: Added a guard to make sure both are `datetime`

**Review by mars-lan — APPROVED** [review]

**Pull request merged**